src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7347,15 +7347,22 @@ private StatementSyntax ParseEmbeddedStatement(bool complexCheck)
                 statement = SyntaxFactory.EmptyStatement(EatToken(SyntaxKind.SemicolonToken));
             }
 
-            // An "embedded" statement is simply a statement that is not a labelled
-            // statement or a declaration statement.
             switch (statement.Kind)
             {
+                // An "embedded" statement is simply a statement that is not a labelled
+                // statement or a declaration statement.
                 case SyntaxKind.LabeledStatement:
                 case SyntaxKind.LocalDeclarationStatement:
                 case SyntaxKind.LocalFunctionStatement:
                     statement = this.AddError(statement, ErrorCode.ERR_BadEmbeddedStmt);
                     break;
+                // Expression statements as embedded statements should be followed by semicolon.
```

Comment:
The comment is incorrect in a very confusing way. An expression statement, according to the language specification, is a statement-expression followed by a semicolon. It doesn't matter if it is embedded or not. An expression statement *always* has a semicolon as part of its syntax.

We have a loosened parser in Interactive mode, which treats the `;` as optional at the EOF. That is kind of horrible. In my opinion, in interactive contexts, at the top level, we should be parsing for an expression or a statement. And if we see an expression followed by an EOF, in that particular case we should treat the semicolon as missing (but not an error) and construct an expression-statement with the missing semicolon. In other words, the approach to fix the issue in this PR is, I believe, not the best way to fix the underlying design error.

Having said that, the approach to the fix in this PR is much less invasive to the parser than the "correct" fix that I describe in the previous paragraph. Given that, I think it makes sense to consider a fix based on this approach.